internal/controllers/clustertemplate_controller.go

```diff
@@ -239,6 +253,69 @@ func validateConfigmapReference[T any](
 	return nil
 }
 
+// validateName return true if the ClusterTemplate name is the
+// format: <name>.<version>, false otherwise
+func validateName(c client.Client, name, version, metadataName, namespace string) error {
+	if metadataName != name+"."+version {
+		return utils.NewInputError("failed to validate ClusterTemplate name %s, should be in the format <spec.name>.<spec.version>: %s", metadataName, name+"."+version)
+	}
+
+	allClusterTemplates := &oranv1alpha1.ClusterTemplateList{}
+	err := c.List(context.Background(), allClusterTemplates)
+	if err != nil {
+		return fmt.Errorf("could not get list of ClusterTemplate across the cluster: %w", err)
+	}
+
+	sameNameVersion := map[string]bool{}
+	sameMetadataName := map[string]bool{}
+	for _, aClusterTemplate := range allClusterTemplates.Items {
+		if aClusterTemplate.Namespace == namespace {
+			continue
+		}
+		if aClusterTemplate.Name == metadataName {
+			sameMetadataName[aClusterTemplate.Namespace] = true
```

Comment:
I think we can return with an error here directly.
Same for the other if.

Reply:
My intent here was to return all the offending objects so that by looking at the message it is easy to fix/delete the other objects